File: backend/Services/Parsers/BracketCityParser.cs
using System.Text.RegularExpressions;

namespace backend.Services.Parsers
{
    public class BracketCityParser : BasicScoreResultParser
    {
        private ILogger<BracketCityParser> _logger;

        public BracketCityParser(ILogger<BracketCityParser> logger) : base(logger)
        {
            _logger = logger;
        }

        public override bool GolfScoring => false;
        public override bool CountWinner => true;
        public override string GameName => "Bracket City";
        public override string? HelpText => null;
        protected override Regex Parser => new Regex($"\\[{GameName}\\].+?Total Score: (?<{ScoreGroup}>\\d+)\\.\\d+", RegexOptions.Singleline);
        protected override string? ExtraContent => null;
        public override string Url => "https://www.theatlantic.com/games/bracket-city";
    }
}

File: lambdas/src/parsers/bracketcity-parser.ts
import { BasicScoreResultParser } from './base/basic-score-parser';

export class BracketCityParser extends BasicScoreResultParser {
  readonly countWinner = true;
  readonly gameName = 'Bracket City';
  readonly golfScoring = false;
  readonly helpText = null;
  protected readonly extraContent = null;
  readonly url = 'https://www.theatlantic.com/games/bracket-city';
  
  // Matches "[Bracket City]\n...\nTotal Score: 91.0" - C# uses brackets and "Total Score:" pattern
  protected readonly parser = new RegExp(
    `\\[${this.gameName}\\].+?Total Score: (?<${BasicScoreResultParser.SCORE_GROUP}>\\d+)\\.\\d+`,
    's'
  );
}
